### PR TITLE
[#IOPID-1832] pagopa user endpoint

### PR DESCRIPTION
**Pull request by arcogabbo**
#### List of Changes

* [[#IOPID-1832] migrated logic of pagopa user endpoint](https://github.com/pagopa/io-auth-n-identity-domain/commit/076926f29729f2a752bc33642534bef132c0f72f)
* [[#IOPID-1832] migrated endpoint unit tests](https://github.com/pagopa/io-auth-n-identity-domain/commit/ba106dffecb4bf6f5bd5be6000555d363d195c65)

#### Motivation and Context

migrate pagopa user endpoint

#### How Has This Been Tested?

migrated unit tests

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/session-manager/api/pagopa.yaml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+parameters: {}
+consumes:
+  - application/json
+produces:
+  - application/json
+securityDefinitions:
+  Token:
+    type: apiKey
+    name: token
+    in: query
```
Comment (BurnedMarshal) [inline apps/session-manager/api/pagopa.yaml:86]: Maybe there is an error on this definition. Looking the library used to extract the token from the request, seems that `access_token` is the right query prop.

_from `passport_http_bearer/lib/strategy.js`:_
```javascript
if (req.query && req.query.access_token) {
  if (token) { return this.fail(400); }
  token = req.query.access_token;
}
```
The library support header, body and query params to transport the token, we should consider to declare in OpenAPI a more safe way, like header.

File: apps/session-manager/src/controllers/pagopa.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  | IResponseErrorNotFound
+  | IResponseErrorTooManyRequests
+  | IResponseSuccessJson<PagoPAUser>
+>;
+
+export const getUser: PagoPAGetUserHandler = (deps) => {
+  const getProfileAndSaveNoticeEmailCache = pipe(
+    ProfileService.getProfile(deps),
+    // TODO: remove this error remap? it was found in io-backend
+    TE.mapLeft((_) => Error("Internal server error")),
```
Comment (BurnedMarshal) [inline apps/session-manager/src/controllers/pagopa.ts:40]: In the backend the `getProfile` returns a `Promise` and the error maps the throw scenario. In here we already have an error on the left path. Maybe do we want to change the error message?

File: apps/session-manager/src/controllers/pagopa.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        ),
+        TE.map((successResponse) => {
+          // if no validated email is provided into InitializedProfile
+          // spid_email will be used for notice email
+          const maybeNoticeEmail: EmailAddress | undefined =
+            successResponse.value.email &&
+            successResponse.value.is_email_validated
+              ? successResponse.value.email
+              : deps.user.spid_email;
+          return O.fromNullable(maybeNoticeEmail);
```
Comment (BurnedMarshal) [inline apps/session-manager/src/controllers/pagopa.ts:73]: We should evaluate this scenario accordingly with uniqueness email address privacy constraint.
cc. @gquadrati

**Review by BurnedMarshal — CHANGES_REQUESTED** [review]

File: apps/session-manager/api/pagopa.yaml (hunk 1 above)
Comment (arcogabbo) [reply apps/session-manager/api/pagopa.yaml]: addressed in https://github.com/pagopa/io-auth-n-identity-domain/pull/55/commits/2a7137aa0ebbcac12991c1d833edfbf2bbb2e39d

File: apps/session-manager/src/controllers/pagopa.ts (hunk 2 above)
Comment (arcogabbo) [reply apps/session-manager/src/controllers/pagopa.ts:40]: I left the same error message with a comment in https://github.com/pagopa/io-auth-n-identity-domain/pull/55/commits/b46301c88ae27cb4314acaf32e81c678ce5bd444

File: apps/session-manager/src/controllers/pagopa.ts (hunk 3 above)
Comment (arcogabbo) [reply apps/session-manager/src/controllers/pagopa.ts]: changed in https://github.com/pagopa/io-auth-n-identity-domain/pull/55/commits/b46301c88ae27cb4314acaf32e81c678ce5bd444

**Review by BurnedMarshal — APPROVED** [review]

**Pull request merged**